onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -189,6 +195,58 @@ namespace py = pybind11;
 using namespace onnxruntime;
 using namespace onnxruntime::logging;
 
+// Custom op section starts
+static Env& platform_env = Env::Default();
+
+CustomOpLibrary::CustomOpLibrary(const char* library_path, OrtSessionOptions& ort_so) {
+  {
+    OrtPybindThrowIfError(platform_env.LoadDynamicLibrary(library_path, &library_handle_));
+
+    if (!library_handle_)
+      throw std::runtime_error("RegisterCustomOpsLibrary: Failed to load library");
+
+    OrtStatus*(ORT_API_CALL * RegisterCustomOps)(OrtSessionOptions * options, const OrtApiBase* api);
+
+    OrtPybindThrowIfError(platform_env.GetSymbolFromLibrary(library_handle_, "RegisterCustomOps", (void**)&RegisterCustomOps));
+
+    if (!RegisterCustomOps)
+      throw std::runtime_error("RegisterCustomOpsLibrary: Entry point RegisterCustomOps not found in library");
+
+    auto* status_raw = RegisterCustomOps(&ort_so, OrtGetApiBase());
+    // Manage the raw Status pointer using a smart pointer
+    auto status = std::unique_ptr<OrtStatus>(status_raw);
+
+    // A non-nullptr indicates status indicates some error
+    if (status) {
+      // TODO: How to handle unload failure ?
+      // Currently we ignore the returned status assuming it is successful
+      platform_env.UnloadDynamicLibrary(library_handle_);
+
+      // Construct error message string
+      std::string error_string = status->msg;
+
+      // Throw
+      throw std::runtime_error(error_string);
+    }
+  }
+}
+
+// Unload the library when the destructor is triggered
+CustomOpLibrary::~CustomOpLibrary() {
+  UnloadLibrary();
```

Comment:
You should never throw in a destructor. Try/catch and log the error.

Reply:
Fixed.